modules/types_protobuf/tests/serialization_tests.cpp

```diff
@@ -0,0 +1,47 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#include <gtest/gtest.h>
+
+#include "hephaestus/random/random_generator.h"
+#include "hephaestus/serdes/protobuf/concepts.h"
+#include "hephaestus/serdes/serdes.h"
+#include "hephaestus/types_protobuf/dummy_type.h"
+
+// NOLINTNEXTLINE(google-build-using-namespace)
+using namespace ::testing;
+
+namespace heph::types::tests {
+
+struct TestTag {};
+
+template <class T>
+class SerializationTests : public ::testing::Test {};
+using SerializationImplementations = ::testing::Types<DummyType>;
+TYPED_TEST_SUITE(SerializationTests, SerializationImplementations);
+
+TYPED_TEST(SerializationTests, TestEmptySerialization) {
```

Comment:
as mentioned before, we went the other way around, creating a random proto, which can be done with reflection without the whole random module, and then deserialize it and serialize it again. Should give the same amount of insights (still only one side completeness is guarateed) with way less customized code

Reply:
Let's discuss your approach in our meeting. I like the idea. The best code is no code, right?

Reply:
We concluded that both random in cpp and random in proto are needed to achieve full test coverage for missing fields.